instituto-brasileiro-de-informacao-em-ciencia-e-tecnologia-abnt.csl

```diff
@@ -291,104 +358,261 @@
     <choose>
-      <if type="article article-journal article-magazine article-newspaper" match="any">
-        <choose>
-          <if variable="volume" match="any">
-            <text term="volume" form="short"/>
-            <text variable="volume" prefix=" " suffix=","/>
-          </if>
-        </choose>
-        <choose>
-          <if variable="issue" match="any">
-            <text term="issue" form="short" prefix=" "/>
-            <text variable="issue" prefix=" "/>
-          </if>
-        </choose>
-        <choose>
-          <if variable="edition" match="any">
-            <text term="edition" form="short" prefix=" " suffix="."/>
-            <text variable="edition" prefix=" "/>
-          </if>
-        </choose>
-        <choose>
-          <if variable="section" match="any">
-            <text term="section" form="short" prefix=", "/>
-            <text variable="section" prefix=" "/>
-          </if>
-        </choose>
+      <if type="book chapter paper-conference report" match="any">
         <choose>
-          <if variable="page" match="any">
-            <text term="page" form="short" prefix=", "/>
-            <text variable="page" prefix=" "/>
+          <if variable="number-of-pages" match="any">
+            <group delimiter=" ">
+              <text variable="number-of-pages"/>
+              <text term="page" form="short"/>
+            </group>
           </if>
         </choose>
       </if>
-      <else-if type="book chapter paper-conference" match="any">
-        <choose>
-          <if variable="volume" match="any">
-            <text term="volume" form="short"/>
-            <text variable="volume" prefix=" " suffix=", "/>
-          </if>
-        </choose>
+      <else-if type="thesis" match="any">
         <choose>
-          <if variable="page" match="any">
-            <text term="page" form="short"/>
-            <text variable="page" prefix=" "/>
+          <if variable="number-of-pages" match="any">
+            <group delimiter=" ">
+              <text variable="number-of-pages"/>
+              <text term="folio" form="short"/>
+            </group>
           </if>
         </choose>
       </else-if>
     </choose>
   </macro>
-  <macro name="citation-locator">
-    <group>
-      <label variable="locator" form="short"/>
-      <text variable="locator" prefix=" "/>
-    </group>
-  </macro>
-  <macro name="genre">
-    <text variable="genre"/>
-  </macro>
-  <macro name="abstract">
-    <text variable="abstract"/>
+  <macro name="performer">
+    <text term="performer" form="long" text-case="capitalize-first" suffix=": "/>
+    <names variable="author">
+      <name delimiter=", " sort-separator=" " delimiter-precedes-last="always">
+        <name-part name="given"/>
+        <name-part name="family"/>
+      </name>
+      <et-al font-style="italic"/>
+    </names>
   </macro>
   <macro name="place">
     <choose>
       <if variable="event-place publisher-place" match="any">
```

Comment:
Can you explain what you are trying to do here?
I am not sure why you're testing for event-place, but then actually don't do anything with it.